docs/data/data-grid/row-grouping/row-grouping.md

```diff
@@ -323,36 +338,34 @@ type GridRowSelectionPropagation = {
 };
 ```
 
-When `rowSelectionPropagation.descendants` is set to `true`.
+### Row selection propagation
 
-- Selecting a parent selects all its filtered descendants automatically.
-- Deselecting a parent row deselects all its filtered descendants automatically.
+When `rowSelectionPropagation.descendants` is set to `true`:
 
-When `rowSelectionPropagation.parents` is set to `true`.
+- Selecting a parent selects all of its filtered descendants automatically
+- Deselecting a parent row deselects of all its filtered descendants automatically
 
-- Selecting all the filtered descendants of a parent selects the parent automatically.
-- Deselecting a descendant of a selected parent deselects the parent automatically.
+When `rowSelectionPropagation.parents` is set to `true`:
 
-The example below demonstrates the usage of the `rowSelectionPropagation` prop.
+- Selecting all the filtered descendants of a parent also selects the parent automatically
+- Deselecting a descendant of a selected parent also deselects the parent automatically
+
+The example below demonstrates the usage of the `rowSelectionPropagation` prop—use the checkboxes at the top to see how the selection behavior changes between parents and children.
 
 {{"demo": "RowGroupingPropagateSelection.js", "bg": "inline", "defaultCodeOpen": false}}
 
-:::info
-The row selection propagation also affects the "Select all" checkbox like any other group checkbox.
-:::
-
-:::info
-The selected rows that do not pass the filtering criteria are automatically deselected when the filter is applied. Row selection propagation is not applied to the unfiltered rows.
-:::
+Row selection propagation also affects the **Select all** checkbox like any other checkbox group.
+Selected rows that do not pass the filtering criteria are automatically deselected when the filter is applied.
+Row selection propagation is not applied to the unfiltered rows.
 
 :::warning
-There are a few limitations to the row selection propagation:
+Row selection propagation has some limitations:
 
-- If `props.disableMultipleRowSelection` is set to `true`, the row selection propagation doesn't apply.
+- If `props.disableMultipleRowSelection` is set to `true`, then row selection propagation won't apply.
 
-- Row selection propagation is a client-side feature and is not supported with the [server-side data source](/x/react-data-grid/server-side-data/).
+- Row selection propagation is a client-side feature and does not support [server-side data sources](/x/react-data-grid/server-side-data/).
```

Comment:
We never use plural of data source elsewhere, could be confused with the actual server side data sources rather than Data Grid's "data source" feature.
```suggestion
- Row selection propagation is a client-side feature and does not support [server-side data](/x/react-data-grid/server-side-data/).
```